core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -732,17 +732,7 @@ class DynamicLogConfig(logManager: LogManager, server: KafkaBroker) extends Brok
   override def reconfigure(oldConfig: KafkaConfig, newConfig: KafkaConfig): Unit = {
     val originalLogConfig = logManager.currentDefaultConfig
     val originalUncleanLeaderElectionEnable = originalLogConfig.uncleanLeaderElectionEnable
-    val newBrokerDefaults = new util.HashMap[String, Object](originalLogConfig.originals)
-    newConfig.valuesFromThisConfig.forEach { (k, v) =>
-      if (DynamicLogConfig.ReconfigurableConfigs.contains(k)) {
```

Comment:
It has been a long time, so I may not remember correctly. But I think we used to only update configs which are `ReconfigurableConfigs` even if you updated ZooKeeper directly with new configs that included others. Non-reconfigurable configs would be picked up only on the next broker restart. Do we think it is safe to include all configs now either because we don't allow other configs to be updated in KRaft or because all log configs are reconfigurable now?

Reply:
Thanks for this comment, @rajinisivaram . I think you're right that we should be excluding changes to configurations that don't appear in `DynamicLogConfig.ReconfigurableConfigs`. I have added some code to do this now. I think from a practical point of view, this only affects `message.format.version`, which we explicitly excluded from being dynamically reconfigurable.

> Do we think it is safe to include all configs now either because we don't allow other configs to be updated in KRaft or because all log configs are reconfigurable now?

KRaft has the same behavior as ZK, in that it lets you set any broker configuration you want, whether it's valid or not. There were too many people depending on this for us to change it in 3.x. Maybe in the future if someone creates a KIP...

Of course the command-line tool does its own validation.

Reply:
Good call @rajinisivaram !